### PR TITLE
HousesServiceImplTests

### PR DESCRIPTION
**Pull request by JEdilbertoTC**

### CODE REVIEW

File: backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+        street.setId("1");
+        street.setName("name 1");
+
+        Mockito.when(housesRepository.save(house)).thenReturn(house);
+        Mockito.when(streetRepository.findById(house.getStreet())).thenReturn(java.util.Optional.of(street));
+        Mockito.when(housesRepository.findById("1")).thenReturn(java.util.Optional.of(house));
+
+        // When:
+        housesService.toggleChipStatusRequest("1", false);
```
Comment (iuresti) [inline backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java:134]: Test only one of the methods ... call getHouseInfo doesn't help to test toggleChipStatusRequest as you don't really use a real repository, it is a mocked one. 

In order to test toggleChipStatusRequest it is enough to test interactions

File: backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        house.setChipsEnabled(true);
+        houses.add(house);
+
+        Street street = new Street();
+
+        street.setId("1");
+        street.setName("name 1");
+
+        // When:
+        Mockito.when(housesRepository.findAllByStreetOrderByNumber("123")).thenReturn(houses);
```
Comment (iuresti) [inline backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java:170]: The "When" section is used to locate the method that you're testing. 

All these sentences "Mockito.when" are part of test initialization and belongs to section "given".

In this case the only line that belongs to "when" section is 
```
HouseInfo houseInfo = housesService.getHouseInfo("1");
```

File: backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
```
Comment (iuresti) [inline backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java:208]: same here with moving "Mockito.when" sentences to "Given" section and "housesService.saveNotes" to "When"

**Review by iuresti — CHANGES_REQUESTED** [review]

File: backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java (hunk 3 above)
Comment (JEdilbertoTC) [reply backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java]: done

File: backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java (hunk 2 above)
Comment (JEdilbertoTC) [reply backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java]: done

File: backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java (hunk 1 above)
Comment (JEdilbertoTC) [reply backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java:150]: done

**Review by iuresti — APPROVED** [review]

**Pull request merged**